src/Avalonia.Controls/Platform/Screen.cs

```diff
@@ -54,12 +104,103 @@ public class Screen
         /// <param name="bounds">The overall pixel-size of the screen.</param>
         /// <param name="workingArea">The actual working-area pixel-size of the screen.</param>
         /// <param name="isPrimary">Whether the screen is the primary one.</param>
+        [Unstable(ObsoletionMessages.MayBeRemovedInAvalonia12)]
         public Screen(double scaling, PixelRect bounds, PixelRect workingArea, bool isPrimary)
         {
-            this.Scaling = scaling;
-            this.Bounds = bounds;
-            this.WorkingArea = workingArea;
-            this.IsPrimary = isPrimary;
-        } 
+            Scaling = scaling;
+            Bounds = bounds;
+            WorkingArea = workingArea;
+            IsPrimary = isPrimary;
+        }
+
+        [PrivateApi]
+        protected Screen(IPlatformHandle platformHandle)
+        {
+            _platformHandle = platformHandle;
+        }
+
+        /// <summary>
+        /// Tries to get the platform handle for the Screen.
+        /// </summary>
+        /// <returns>
+        /// An <see cref="IPlatformHandle"/> describing the screen handle, or null if the handle
+        /// could not be retrieved.
+        /// </returns>
+        public IPlatformHandle? TryGetPlatformHandle() => _platformHandle;
+        
+        /// <inheritdoc/>
+        public override bool Equals(object? obj)
+        {
+            if (obj is null) return false;
+            if (ReferenceEquals(this, obj)) return true;
+            if (obj.GetType() != GetType()) return false;
+            return Equals((Screen)obj);
+        }
+
+        /// <inheritdoc/>
+        public override int GetHashCode()
+        {
+            // ReSharper disable NonReadonlyMemberInGetHashCode
+            return _platformHandle?.GetHashCode()
+                   ?? (IsPrimary, DisplayName, CurrentOrientation, Scaling, Bounds, WorkingArea).GetHashCode();
```

Comment:
I believe that we aren't allowed to change the hash code of a class at runtime. Screen class, however, is now mutable.

Reply:
I will just return `base.GetHashCode()`, when platform handle is not avaialble. This code path is only executed on Linux and for users who manually created Screen object (for some reason that's allowed).

Reply:
Moved these hashcode/equals implementation to another PlatformScreen type.